openshift-pipelines/catalog-cd
# Added Logic for testing name conflicts from various sources

This PR is meant to add a precursor check for name conflicts of various kinds ( tasks or pipelines ) before using the externals subcommand.

How to test:
so you can create an externals.yaml as follows
```
repositories:
  - url: https://github.com/Aneesh-M-Bhat/test-release-1
    types:
      - tasks
      - pipelines
  - url: https://github.com/Aneesh-M-Bhat/test-release-2
    types:
      - tasks
      - pipelines
  - url: https://github.com/Aneesh-M-Bhat/test-release-3
    types:
      - tasks
      - pipelines
```
keep it in the same directory as the project & then run 
```
make build
./catalog-cd catalog externals --config="./externals.yaml"
```
comment out the above externals items to see the various possibilities
